### PR TITLE
docs: document new `suggest-tests` tool

### PR DESCRIPTION
**Pull request by Ezrashaw**
Part of rust-lang/rust#109933.
Blocked on rust-lang/rust#106249? (the links are broken until it's merged)

I think it's a pretty good overview, both high and low level.

r? @albertlarsan68

### CODE REVIEW

File: src/tests/suggest-tests.md [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+The following steps should serve as a rough guide to add suggestions to
+`suggest-tests` (very welcome!):
+
+1. Determine the rules for your suggestion. Is it simple and operates only on
+   a single path or does it match globs? Does it need fine-grained control over
+   the resulting command or does "one size fit all"?
+2. Based on the previous step, decide if your suggestion should be implemented
+   as either static or dynamic.
+3. Implement the suggestion. If it is dynamic then a test is required, see the
```
Comment (albertlarsan68) [inline src/tests/suggest-tests.md:51]: Is the fact that the tests are required enforced somewhere (Lint/check?) Not a blocker, but would be good as a follow-up. In the meantime, say something like
```suggestion
3. Implement the suggestion. If it is dynamic then a test is highly recommended, to verify that your logic is correct and to give an example of the suggestion. See the
```

**Review by albertlarsan68 — COMMENTED** [review]
Overall great, a few typos and a bigger question

File: src/tests/suggest-tests.md (hunk 1 above)
Comment (Ezrashaw) [reply src/tests/suggest-tests.md]: Good idea. I'll add it to the tracking issue. Not entirely sure how it could be implmented though.

**Comment by Ezrashaw** [comment]
@albertlarsan68 Implemented. I was far too tired when I wrote and proofread this :P.

**Comment by Ezrashaw** [comment]
@rustbot label -blocked

@albertlarsan68 This can now be merged, the main PR has been merged.

File: src/tests/suggest-tests.md [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+`suggest-tests` (very welcome!):
+
+1. Determine the rules for your suggestion. Is it simple and operates only on
+   a single path or does it match globs? Does it need fine-grained control over
+   the resulting command or does "one size fit all"?
+2. Based on the previous step, decide if your suggestion should be implemented
+   as either static or dynamic.
+3. Implement the suggestion. If it is dynamic then a test is highly recommended,
+   to verify that your logic is correct and to give an example of the suggestion.
+   See the [tests.rs](https://github.com/rust-lang/rust/blob/master/src/tools/suggest-tests/src/static_suggestions.rs)
```
Comment (tshepang) [inline src/tests/suggest-tests.md:53]:
```suggestion
   See the [tests.rs](https://github.com/rust-lang/rust/blob/master/src/tools/suggest-tests/src/tests.rs)
```

**Review by tshepang — APPROVED** [review]

**Pull request merged**